cheroot/test/test_ssl.py

```diff
@@ -757,3 +799,142 @@ def test_http_over_https_error(
+        'builtin-encrypted-key-with-bytes-password',
+        'builtin-non-encrypted-key-with-str-password',
+        'builtin-non-encrypted-key-with-bytes-password',
+        'pyopenssl-encrypted-key-with-str-password',
+        'pyopenssl-encrypted-key-with-bytes-password',
+        'pyopenssl-non-encrypted-key-with-str-password',
+        'pyopenssl-non-encrypted-key-with-bytes-password',
+    ),
+)
+# pylint: disable-next=too-many-positional-arguments
+def test_ssl_adapters_with_private_key_password(
+    private_key_password,
+    tls_certificate_chain_pem_path,
+    tls_certificate_passwd_private_key_pem_path,
+    tls_certificate_private_key_pem_path,
+    adapter_type,
+    encrypted_key,
+    password_as_bytes,
+):
+    """Check that server starts using ssl adapter with password-protected private key."""
+    httpserver = HTTPServer(
+        bind_addr=(ANY_INTERFACE_IPV4, EPHEMERAL_PORT),
+        gateway=Gateway,
+    )
+
+    key_file = (
+        tls_certificate_passwd_private_key_pem_path
+        if encrypted_key
+        else tls_certificate_private_key_pem_path
+    )
+    key_pass = (
+        private_key_password.encode('utf-8')
+        if password_as_bytes
+        else private_key_password
+    )
+
+    tls_adapter_cls = get_ssl_adapter_class(name=adapter_type)
+    tls_adapter = tls_adapter_cls(
+        certificate=tls_certificate_chain_pem_path,
+        private_key=key_file,
+        private_key_password=key_pass,
+    )
+
+    httpserver.ssl_adapter = tls_adapter
+    httpserver.prepare()
+
+    assert httpserver.ready
+    assert httpserver.requests._threads
+    for thr in httpserver.requests._threads:
+        assert thr.ready
+
+    httpserver.stop()
+
+
+@pytest.mark.parametrize(
+    'adapter_type',
+    ('builtin',),
+)
+def test_builtin_adapter_with_false_key_password(
+    tls_certificate_chain_pem_path,
+    tls_certificate_passwd_private_key_pem_path,
+    adapter_type,
+):
+    """Check that builtin ssl-adapter initialization fails when wrong private key password given."""
+    tls_adapter_cls = get_ssl_adapter_class(name=adapter_type)
+    with pytest.raises(ssl.SSLError, match=r'\[SSL\] PEM.+'):
+        tls_adapter_cls(
+            certificate=tls_certificate_chain_pem_path,
+            private_key=tls_certificate_passwd_private_key_pem_path,
+            private_key_password='x' * 256,
+        )
+
+
+@pytest.mark.parametrize(
+    ('adapter_type', 'false_password', 'expected_warn'),
+    (
+        (
+            'pyopenssl',
+            '837550fd-bcb9-4320-87e6-09de6456b09',
+            contextlib.nullcontext(),
+        ),
+        ('pyopenssl', 555555, contextlib.nullcontext()),
+        (
+            'pyopenssl',
+            '@' * 2048,
+            pytest.warns(
+                UserWarning,
+                match=r'User-provided password is 2048 bytes.+',
+            ),
+        ),
+    ),
+    ids=('false-password', 'integer-password', 'too-long-password'),
```

Comment:
I think "incorrect" is easier to put in the context than "false". At least, it sounds more obvious to me.
```suggestion
    ids=('incorrect-password', 'integer-password', 'too-long-password'),
```